calc/calc.go

```diff
@@ -0,0 +1,51 @@
+package calc
+
+import (
+	"errors"
+)
+
+type calculator struct {
+}
+
+func NewCalculator() *calculator {
+	return &calculator{}
+}
+
+func (c calculator) Calculate(n1 float64, m string, n2 float64) (float64, error) {
+	switch m {
+	case "+":
+		return add(n1, n2), nil
+	case "-":
+		return sub(n1, n2), nil
+	case "*":
+		return mul(n1, n2), nil
+	case "/":
+		ans, err := div(n1, n2)
+		if err != nil {
+			return 0, err
+		} else {
+			return ans, nil
+		}
```

Comment:
рекомендую не использовать else, там где можно его упростить
```suggestion
		if err != nil {
			return 0, fmt.Errorf("division: %v",err)
		} 
	
	        return ans, nil	
```